lgc/include/lgc/state/TargetInfo.h

```diff
@@ -138,6 +138,23 @@ class TargetInfo {
   // Set TargetInfo. Returns false if the GPU name is not found or not supported.
   bool setTargetInfo(llvm::StringRef gpuName);
 
+  // GFX IP checkers
+  bool isGfx(unsigned major, unsigned minor = 0, unsigned stepping = 0) const { // GFX(IP)
+    return m_gfxIp.major == major && m_gfxIp.minor == minor && m_gfxIp.stepping == stepping;
+  }
+  bool isGfxAbove(unsigned major, unsigned minor = 0, unsigned stepping = 0) const { // > GFX(IP)
+    if (m_gfxIp.major > major || (m_gfxIp.major == major && m_gfxIp.minor > minor) ||
+        (m_gfxIp.major == major && m_gfxIp.minor == minor && m_gfxIp.stepping > stepping))
```

Comment:
Consider using `std::tie` to get these comparisons for free and 100% correct.
Something like `std::tie(m_gfxIp.major, m_gfxIp.mino, steping) > std::tie(major, minor, stepping)`.

Reply:
Thanks. It is a good idea.